src/util/FactorUtil.js

```diff
@@ -155,8 +157,15 @@ function (Okta, TimeUtil) {
       return 'QUESTION';
     }
     if (provider === 'FIDO' && factorType === 'webauthn') {
-      return 'WINDOWS_HELLO';
-    }
+      if (this.settings != null && this.settings.get('features.webauthn') != null) {
+        webauthnEnabled = this.settings.get('features.webauthn');
```

Comment:
hmm. upon successful primary auth, this function is called in two paths: one when `Factor.factorName` is being derived, and one when `Factor.factorLabel` model is derived, it doesn't seem to have access to `this.settings` for the second path, any advice?

Reply:
Haven't dug into the code, but wouldn't your description be a problem? I.e. in one of those paths, for the same factor, this will resolve to 'U2F' when it has access to settings, and 'WINDOWS_HELLO'.